cpp/include/nvtext/normalize.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+namespace nvtext
+{
+
+/**
+ * @brief Returns a new strings column by normalizing the whitespace in each
+ * string in the input column.
+ *
+ * Normalizing a string replaces any number of whitespace character runs
+ * with a single space and trims whitespace from the beginning and end
+ * of the string.
+ *
+ * Example:
+ * ```
+ * s = ["a b", "  c  d\n", "e \t f "]
+ * t = normalize_spaces(s)
+ * t is now ["a b","c d","e f"]
+ * ```
```

Comment:
It would be nice to add some details on, what are the whitespace characters (<=32 ascii) & unicode support.
https://en.wikipedia.org/wiki/Whitespace_character